include/picongpu/particles/collision/IntraCollision.hpp

```diff
@@ -29,375 +29,350 @@
+            using FramePtr = typename T_ParBox::FramePtr;
+            detail::cellDensity<FramePtr>(worker, forEachCell, parCellList, densityArray, accFilter);
 
-                    auto accFilter = filter(worker, localSuperCellOffset);
+            worker.sync();
 
-                    /* loop over all particles in the frame */
-                    auto forEachFrameElem = lockstep::makeForEach<frameSize>(worker);
-                    FramePtr firstFrame = pb.getFirstFrame(superCellIdx);
+            // shuffle indices list
+            forEachCell(
+                [&](uint32_t const linearIdx) {
+                    detail::shuffle(
+                        worker,
+                        parCellList.particleIds(linearIdx),
+                        parCellList.size(linearIdx),
+                        rngHandle);
+                });
 
-                    prepareList(
+            auto collisionFunctorCtx = forEachCell(
+                [&](int32_t const idx)
+                {
+                    auto parAccess = parCellList.template getParticlesAccessor<FramePtr>(idx);
+                    uint32_t const sizeAll = parAccess.size();
+                    uint32_t potentialPartners = sizeAll - 1u + sizeAll % 2u;
+                    auto collisionFunctor = srcCollisionFunctor(
                         worker,
-                        forEachFrameElem,
-                        deviceHeapHandle,
-                        pb,
-                        firstFrame,
-                        numParticlesInSupercell,
-                        parCellList,
-                        nppc,
-                        accFilter);
+                        localSuperCellOffset,
+                        densityArray[idx],
+                        densityArray[idx],
+                        potentialPartners);
+                    if(sizeAll >= 2u)
+                    {
+                        if constexpr(useScreeningLength)
+                        {
+                            auto const shifted = screeningLengthSquared.shift(superCellIdx * SuperCellSize::toRT());
+                            auto const idxInSuperCell = pmacc::math::mapToND(SuperCellSize::toRT(), idx);
+                            collisionFunctor.coulombLogFunctor.screeningLengthSquared_m = shifted(idxInSuperCell)[0];
+                        }
+                        for(uint32_t i = 0; i < sizeAll; i += 2)
+                        {
+                            auto parEven = parAccess[i];
+                            auto parOdd = parAccess[(i + 1) % sizeAll];
+                            // In Higginson 2020 eq. (31) s has an additional 1/2 factor for
+                            // intraCollisions (compare with 29 and use m_aa = 1/2 m_a). Bu this seams to be a
+                            // typo smilei doesn't include this extra factor. It was applied here in the
+                            // previous version, via the duplication correction.
+
```

Comment:
How do you know which one is correct?

Reply:
Wenn ich mich richtig erinnere, dann habe ich das auch nachgerechnet.